packages/admin/src/middleware.page.ts

```diff
@@ -54,9 +54,17 @@ export async function middleware(req: NextRequest) {
     });
 
     res.headers.set('Cache-Control', 'no-store');
-
     return res;
   } else {
-    return NextResponse.redirect(getLoginUrl());
+    if (submissionDownloadPattern.test({ pathname: req.nextUrl.pathname })) {
+      const url = getLoginUrl() + req.nextUrl.pathname;
+      return NextResponse.redirect(url);
+    } else {
+      return NextResponse.redirect(getLoginUrl());
+    }
```

Comment:
could simplify the logic here:
```suggestion
    let url = getLoginUrl()
    if (submissionDownloadPattern.test({ pathname: req.nextUrl.pathname })) {
      url = url + req.nextUrl.pathname;
    }
    return NextResponse.redirect(url);
```